aiopg/connection.py

```diff
@@ -421,3 +420,27 @@ def __del__(self):
     def notifies(self):
         """Return notification queue."""
         return self._notifies
+
+
+class _CursorContextManager:
```

Comment:
That's some really interesting code.  Thanks!

To make sure we're clear, what's the goal here?  In my test below, I assert `asyncio.iscoroutine(conn.cursor())`; is this the goal?  What about `asyncio.iscoroutinefunction(conn.cursor))`?  (As far as I can tell, aiohttp does not achieve this.)  Are there any other requirements?

aiohttp [modifies asyncio.coroutines._COROUTINE_TYPES](https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/client.py#L465), which is relying on the structure of CPython implementation.  You chastised me for doing similar previously, what's the difference?  Why is `coroutines._COROUTINE_TYPES += (_RequestContextManager,)` okay, but `cursor._is_coroutine = True` bad?  (Just curious, I'm not offended at having to remove `cursor._is_coroutine`.  It was a last resort to get `asyncio.iscoroutinefunction` to behave.)